lib/block_view_types.go

```diff
@@ -4647,6 +4647,43 @@ func DecodeByteArray(reader io.Reader) ([]byte, error) {
 	}
 }
 
+func EncodeUint64Array(uint64s []uint64) []byte {
+	var data []byte
+
+	data = append(data, UintToBuf(uint64(len(uint64s)))...)
+	for _, uint64 := range uint64s {
+		data = append(data, UintToBuf(uint64)...)
+	}
+
+	return data
+}
+
+func DecodeUint64Array(reader io.Reader) ([]uint64, error) {
+	arrLen, err := ReadUvarint(reader)
+	if err != nil {
+		return nil, errors.Wrapf(err, "DecodeUint64Array: Problem reading array length")
+	}
+
+	if arrLen <= 0 {
+		return nil, nil
+	}
```

Comment:
if arrLen == 0, should we return an empty slice instead of a nil value? Also, a uint can't be less than 0.
```suggestion
	if arrLen == 0 {
		return []uint64{}, nil
	}
```

Reply:
Returning nil matches the decoder pattern elsewhere. Nil slice and an initialized empty slice have the same observable behavior in golang. No practical difference between the two. See https://stackoverflow.com/questions/44305170/nil-slices-vs-non-nil-slices-vs-empty-slices-in-go-language.